android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -389,6 +392,40 @@ private static byte[] takeScreenshotOnUiThread(Activity activity) {
         return bytesWrapper[0];
     }
 
+    public void fetchViewHierarchy(Promise promise) {
+        if (serializer == null) {
+            logger.log(SentryLevel.ERROR, "Could not get ViewHierarchy, serializer is null, likely native sdk has not been initialized.");
+            promise.resolve(null);
+            return;
+        }
+
+        final @Nullable Activity activity = getCurrentActivity();
+        final @Nullable ViewHierarchy viewHierarchy = ViewHierarchyEventProcessor.snapshotViewHierarchy(activity, logger);
```

Comment:
M: What if `ViewHierarchyEventProcessor` already has a method called `snapshotViewHierarchyAsData` or something that returns the `bytes` directly as suggested before, right now the Hybrid SDK needs to take care of serialization and this is the Native SDK's responsibility IMO.
I'd not block this PR because of that, but that was the suggestion before the refactoring, wondering why we've not taken this approach.

Reply:
That related [to the first comment](https://github.com/getsentry/sentry-react-native/pull/2708/files/f64ac6fa97a762dba4cb8b21601e73c76154b96e#r1092125493) because I didn't know how to get a reference of the serializer in the ViewHierarchyEventProcessor.

Reply:
I've opened PR in java https://github.com/getsentry/sentry-java/pull/2508

But as you said we can finish this and when the java PR is released we can simplify it in a new PR?

Reply:
Fine for me, just address the other comments.